### PR TITLE
Rem 19 rest api

### PR DESCRIPTION
**Pull request by anatolii15**
REM-46 REST API: Make proper http error codes more common
REM-45 REST API: Make proper model validation for certificates

Main validation now located at certificate_api_decorator.py
The idea of it is separation of the controllers logic (revoke, store, check)
and additional validation/creation logic.

### CODE REVIEW

File: remme/rest_api/certificate.py [hunk 1]
```diff
@@ -14,111 +14,53 @@
...
-def _make_address_from_pem(client, certificate_pem):
-    certificate = x509.load_pem_x509_certificate(certificate_pem.encode('utf-8'),
-                                                 default_backend())
-    crt_bin = certificate.public_bytes(serialization.Encoding.DER).hex()
-    address = client.make_address_from_data(crt_bin)
-    return address
-
-
-def post(payload):
+@http_payload_required('certificate')
```
Comment (eugene-babichenko) [inline remme/rest_api/certificate.py]: Totally not needed. Connexion performs the validation of this part.

Comment (anatolii15) [reply remme/rest_api/certificate.py]: If you try pass empty json(payload NoneType) or json with no certificate prop(KeyError: 'certificate'), there will be 500 error, so I've added that for explicit error codes and messages binding

Comment (eugene-babichenko) [reply remme/rest_api/certificate.py]: This should be resolved by putting the `required` attribute into the OpenAPI specification.

Comment (anatolii15) [reply remme/rest_api/certificate.py]: Sounds valid. Will try that, thanks

File: remme/rest_api/certificate_api_decorator.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    def func_wrapper(payload):
+        if payload is None:
+            return {'error': 'Http request body can not be empty'}, 422
+
+        return func(payload)
+
+    return func_wrapper
+
+
+# endregion
```
[the remark this answers is not in the document]
Comment (eugene-babichenko) [reply remme/rest_api/certificate_api_decorator.py]: I think it is good old VS style (`#pragma region ...`). Need to be removed

File: remme/rest_api/certificate_api_decorator.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+# region Certificate Creation
+
+def create_certificate(payload):
```
[the remark this answers is not in the document]
Comment (eugene-babichenko) [reply remme/rest_api/certificate_api_decorator.py]: Comments should be here for sure. About naming: those names are fine and the file name is more than descriptive.

**Comment by eugene-babichenko** [comment]
There was a bit of misunderstanding, most of input validation is performed by Connexion based on OpenAPI specs. If something is not, we should report it as a bug, implement a hotfix in our code and make an appropriate notice with the link to the reported bug.

**Comment by bogomazov** [comment]
Is there any more extended validation needed in place, then pattern matching by OpenAPI?

**Comment by eugene-babichenko** [comment]
Not yet, all the other validation is done in `py-cryptography`, the transaction handler and so on.

**Pull request merged**